### PR TITLE
Add basic test suite setup

### LINKED ISSUE #68: Add CI tests for the library
Adding tests / CI was mentioned in #50, but I'm splitting this off into its own issue as I feel like this could be accomplished without also converting the library to TS.

Having a test suite that could be run against the library would be awesome. To accomplish this, my suggestion would be to utilize the [testcontainers](https://www.npmjs.com/package/testcontainers) library to spin up a presto/trino container + postgres container that are networked together, and then can just run through the basics of the existing client methods. With this in place, would be easy to extend over time with more complex configurations.

Happy to contribute to this effort @tagomoris if a PR would be accepted following the above plan (or if you've got some other idea of how to do it).

### PR DESCRIPTION
**Pull request by MasterOdin**
Closes #68 

This PR adds a basic test setup for the repo that'll run for any push to the `master` branch as well as for any PR made against it. Currently the tests used are very minimal, focusing just on getting the basics of what a test suite can look like, with my hope that would add additional tests going forward.

### CODE REVIEW

File: index.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+});
+
+describe.each([
+  ['presto', 'ahanaio/prestodb-sandbox:0.281'],
+  ['trino', 'trinodb/trino:418'],
+])('%s', function(engine, image){
+  var container;
+  var client;
+  beforeAll(function(done){
+    new GenericContainer(image)
```
Comment (MasterOdin) [inline index.spec.js:26]: I waffled between using docker-compose vs testcontainers and settled on testcontainers as it gives more flexibility in terms of setting up tests that require some custom server configuration (e.g. like basic auth), though that is still all possible with docker-compose.

The runtime cost of using testcontainers is that running the test suite takes ~30s, with most of that time taken purely on waiting for the containers to start.

If opposed, easy enough to switch to using docker-compose, let me know what you'd prefer @tagomoris.

**Comment by tagomoris** [comment]
I'm busy just now and can't see the detail of this change. I'll be able to take time for this later this week.
@MasterOdin Please ping me next week if I don't make any comments.

File: index.spec.js (hunk 1 above)
Comment (tagomoris) [reply index.spec.js:26]: I'm totally fine with the library selection 👍

**Comment by tagomoris** [comment]
The code looks good.
But I ran the tests and got timeout errors on my laptop (macOS, Docker desktop is running). Could you check and modify the default timeout or any other issues on running those on laptops?

```
 FAIL  ./index.spec.js (120.813 s)
  ✓ cannot use basic and custom auth (8 ms)
  presto
    ✕ simple query
    ✕ query with error
  trino
    ✕ simple query
    ✕ query with error

  ● presto › simple query

    thrown: "Exceeded timeout of 60000 ms for a hook while waiting for `done()` to be called.
    Add a timeout value to this test to increase the timeout, if this is a long-running test. See https://jestjs.io/docs/api#testname-fn-timeout."

      23 |   var container;
      24 |   var client;
    > 25 |   beforeAll(function(done){
         |   ^
      26 |     new GenericContainer(image)
      27 |       .withExposedPorts(8080)
      28 |       .withWaitStrategy(Wait.forLogMessage('SERVER STARTED'))

      at beforeAll (index.spec.js:25:3)
          at Array.forEach (<anonymous>)
      at Object.<anonymous> (index.spec.js:22:3)
(snip)
```

**Comment by MasterOdin** [comment]
I ended up removing the usage of `testcontainers` in favor of `docker-compose` as I think the image sizes are a bit frustrating to deal with in terms of getting timeouts right, without just letting jest potentially run for a long time. There's also some issues in that since the presto image only comes in `amd64`, it runs kind of poorly on mac M machines, which makes it even harder to get it right.

File: index.spec.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        expect(data).toEqual([[1]]);
+        expect(columns).toHaveLength(1);
+        expect(columns[0]).toEqual(expect.objectContaining({ name: 'col', type: 'integer' }));
+      },
+      callback: function(error){
+        expect(error).toBeNull();
+        done();
+      },
+    });
+  }, 10000);
```
Comment (MasterOdin) [inline index.spec.js:45]: The default jest timeout is 5s. I am usually seeing this test pass in 2-3s, though my coworker sometimes gets timeouts with presto. I suspect this is due to the fact presto is run through amd64 on our M1 machines where trino uses arm64 is a bit snappier.

**Comment by MasterOdin** [comment]
Additionally, see https://github.com/MasterOdin/presto-client-node/actions/runs/5091070215 for what the CI action looks like when it's triggered.

**Comment by tagomoris** [comment]
This is a great addition to this repository. Thank you!

I confirmed this works well on my laptop after writing `docker-compose.yaml`.
@MasterOdin Could you add `docker-compose.yaml` with the content below?

```yaml
version: '3.7'

services:
  presto:
    image: ahanaio/prestodb-sandbox:0.281
    ports:
      - "18080:8080"
  trino:
    image: trinodb/trino:418
    ports:
      - "18081:8080"
```

And a section to README about how to run tests on local environments? It should notify developers to wait the lines below after `docker compose up`:

```
presto-client-node-trino-1   | 2023-06-02T08:12:37.760Z	INFO	main	io.trino.server.Server	======== SERVER STARTED ========
presto-client-node-presto-1  | 2023-06-02T08:13:29.760Z	INFO	main	com.facebook.presto.server.PrestoServer	======== SERVER STARTED ========
```

**Comment by MasterOdin** [comment]
Whoops, totally meant to include the `docker-compose.yml` file, sorry! I've also updated the README with a `## Development` section that includes running the `docker-compose.yml` file as well as running the test suite.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+version: "3.9"
+
+services:
+  presto:
+    image: ahanaio/prestodb-sandbox:0.281
+    # there is no arm64 version of this image
```
Comment (tagomoris) [inline docker-compose.yml:6]: 👍 nice comment!

**Review by tagomoris — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -180,6 +180,22 @@ var client = new presto.Client({
 });
 ```
 
+## Development
+
+When working on this library, you can use the included docker-compose.yml file to spin up a Presto and Trino DBs, which can be done with:
+
+```
+docker compose up
+```
```
Comment (tagomoris) [inline README.md:189]: As I commented (https://github.com/tagomoris/presto-client-node/pull/69#issuecomment-1573343931), I want a note that we should wait for the servers' launch completion:

```
presto-client-node-trino-1   | 2023-06-02T08:12:37.760Z	INFO	main	io.trino.server.Server	======== SERVER STARTED ========
presto-client-node-presto-1  | 2023-06-02T08:13:29.760Z	INFO	main	com.facebook.presto.server.PrestoServer	======== SERVER STARTED ========
```

Comment (MasterOdin) [reply README.md:189]: So sorry, added in 3365b571f1c15b51d8c0e62ad7c6d6a3c172d7b2.

**Comment by tagomoris** [comment]
@MasterOdin Thank you for the contribution!

**Pull request merged**